docs/component/authorization.md

```diff
@@ -0,0 +1,109 @@
+# Resource Based Access Control in Insights Results Smart Proxy
+
+## Overview
+
+The Insights Results Smart Proxy implements an authorization mechanism to control access to its endpoints. This functionality ensures that only authorized users can access certain resources based on their roles and permissions. The authorization is primarily handled through a middleware that checks the user's identity and permissions before allowing access to specific routes.
+
+## Middlewares
+
+### Authentication Middleware
+
+The authentication middleware is responsible for verifying that the request has been made by a known party, and that said party's identity token has all the information necessary for identification of the concrete requester. It is implemented in the `auth` module and the `Authentication` function within the `auth_middleware.go` file. If enabled, authentication always happens before the authorization process.
+
+### Authorization Middleware
+
+The authorization middleware is responsible for enforcing role-based access control (RBAC) for incoming requests. It is implemented in the `auth` module and the `Authorization` function within the `auth_middleware.go` file. The authorization middleware is set up in the `setupAuthMiddleware` function, which configures the router to use the `Authorization` middleware for routes that require RBAC checks.
```

Comment:
I’ve just realized auth_middleware may be misleading as it could refer to authorization or authentication, but I guess it’s not a big deal